qa/tasks/cephfs/test_volume_client.py

```diff
@@ -970,6 +970,27 @@ def test_get_object(self):
             obj_data = obj_data
         )))
 
+    def test_update_object(self):
+        vc_mount = self.mounts[1]
+        vc_mount.umount_wait()
+        self._configure_vc_auth(vc_mount, "manila")
+
+        obj_data = 'test_data'
+        modified_data = 'modified_test_data'
+        obj_name = 'test_vc_ob_2'
+        pool_name = self.fs.get_data_pool_names()[0]
+        self.fs.rados(['put', obj_name, '-'], pool=pool_name, stdin_data=obj_data)
+
+        def data_modifier(data):
+            return 'modified_' + data
+        self._volume_client_python(vc_mount, dedent("""
+            try:
+                vc.update_object("{pool_name}", "{obj_name}", "{data_modifier}")
```

Comment:
This test doesn't really exercise the atomic put. You will want to get a version of an object, update it normally using `put_object`, and then "try" to do another `put_object_versioned` with the version you got in the first step (it should fail).